syntax = "proto3";

package todo;

service Todo {
    rpc ListTask(ListTaskRequest) returns (stream ListTaskResponse);
    rpc GetTaskById(GetTaskByIdRequest) returns (GetTaskByIdResponse);
    rpc GetTaskByName(GetTaskByNameRequest) returns (GetTaskByNameResponse);
}

enum Priority {
    URG = 0;
    HIGH = 1;
    MED = 2;
    LOW = 3;
}

message Task {
    int32 id = 1;
    string name = 2;
    bool is_complete = 3;
    Priority priority = 4;
}

message ListTaskRequest {
}

message ListTaskResponse {
    Task task = 1;
}

message GetTaskByIdRequest {
    int32 id = 1;
}

message GetTaskByIdResponse {
    Task task = 1;
}

message GetTaskByNameRequest {
    string name = 1;
}

message GetTaskByNameResponse {
    Task task = 1;
}
